### PR TITLE
implemented history clearing handler + reset leo confirmation

### PR DESCRIPTION
**Pull request by vadimstruts**
- Added checkboxes to clear Brave Leo Assistant for "Advanced" and "On Exit" tabs
- Implemented common handler for both cases (It is currently doing nothing).
- Added dialog to confirm when you reset Brave Leo prefs to default

Resolves https://github.com/brave/brave-browser/issues/31441

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

-  make sure that new checkboxes are visible on the `brave://settings/clearBrowserData` ("Advanced" and "On Exit" tabs) when you enabled `AI-Chat` and hidden when disabled
-  make sure that confirmation dialog works properly when you click "Clear Leo data" on the `brave://settings/leo-assistant` (It means after confirmation: `Show Leo icon in the sidebar` is ON, `Show suggested prompts in the conversation` is OFF and `AI-Chat` icon is shown on the sidebar, otherwise nothing changed)

### CODE REVIEW

**Comment by vadimstruts** [comment]
It is a duplicate of the PR: https://github.com/brave/brave-core/pull/19504
The original one, has been closed automatically, due to  branch messed up on `rebase`

File: browser/resources/settings/brave_leo_assistant_page/brave_leo_assistant_page.ts [hunk 1]
```diff
@@ -44,7 +44,11 @@ class BraveLeoAssistantPageElement extends BraveLeoAssistantPageBase {
       BraveLeoAssistantBrowserProxyImpl.getInstance()
 
     onResetAssistantData_() {
-      this.browserProxy_.resetLeoData()
+      var message =
```
Comment (fallaciousreasoning) [inline browser/resources/settings/brave_leo_assistant_page/brave_leo_assistant_page.ts:47]: Nit:
```suggestion
      const message =
```

**Review by fallaciousreasoning — APPROVED** [review]
LGTM

**Review by goodov — APPROVED** [review]

**Comment by vadimstruts** [comment]
Merged as got approved by the several devs

**Pull request merged**